x-pack/filebeat/input/awss3/input.go

```diff
@@ -308,12 +315,14 @@ func getRegionFromQueueURL(queueURL string, endpoint string) (string, error) {
 		return "", fmt.Errorf(queueURL + " is not a valid URL")
 	}
 	if url.Scheme == "https" && url.Host != "" {
```

Comment:
Are you setting region name in the config? That is the intention here. Maybe I should add a note in the docs to this effect. We can use a dummy value like "local" if it doesn't actually mean anything to localstack.

Reply:
I have set the region value. But then it failed in the next condition where it checked `regionName != in.config.RegionName`, because the regionName from URL is always `""`.

How does this sound?

``` diff
diff --git a/x-pack/filebeat/input/awss3/input.go b/x-pack/filebeat/input/awss3/input.go
index 9ba508d439..6349a041fb 100644
--- a/x-pack/filebeat/input/awss3/input.go
+++ b/x-pack/filebeat/input/awss3/input.go
@@ -12,6 +12,7 @@ import (
        "strings"
        "time"
 
        awssdk "github.com/aws/aws-sdk-go-v2/aws"
        "github.com/aws/aws-sdk-go-v2/service/s3"
        "github.com/aws/aws-sdk-go-v2/service/sqs"
@@ -115,22 +116,22 @@ func (in *s3Input) Run(inputContext v2.Context, pipeline beat.Pipeline) error {
        defer cancelInputCtx()
 
        if in.config.QueueURL != "" {
-               regionName, err := getRegionFromQueueURL(in.config.QueueURL, in.config.AWSConfig.Endpoint)
-               if err != nil && in.config.RegionName == "" {
-                       return fmt.Errorf("failed to get AWS region from queue_url: %w", err)
-               }
-               if in.config.RegionName != "" && regionName != in.config.RegionName {
-                       inputContext.Logger.Warnf("configured region disagrees with queue_url region: %q != %q: using %[1]q",
-                               in.config.RegionName, regionName)
-                       regionName = in.config.RegionName
+               // If region is configured on the input then use it by default
+               if in.config.RegionName != "" {
+                       in.awsConfig.Region = in.config.RegionName
+               } else {
+                       regionName, err := getRegionFromQueueURL(in.config.QueueURL, in.config.AWSConfig.Endpoint)
+                       if err != nil {
+                               return fmt.Errorf("faaaaaailed to get AWS region from queue_url: %w", err)
+                       }
+                       // set the region name from the queue_url
+                       in.awsConfig.Region = regionName
                }
 
-               in.awsConfig.Region = regionName
```

Reply:
I'll do `if in.config.RegionName != "" && regionName != "" && regionName != in.config.RegionName {`. I want to confirm that we are never in a state where there is an ambiguous region name.

Reply:
Sounds good. Can we have some unit testing covering different scenarios?

Reply:
Not in this section of code. That's what your localstack test is for.

Reply:
Ideally we could have an integration test using localstack.

Reply:
The idea is to introduce an integration test suite using localstack alongside real aws resource testing. I was thinking of unit testing as there are multiple `if` conditions and localstack does not cover all the scenarios

- in.config.RegionName = ""
- regionName from `queue_url` is "" - localstack can only do this
- url does not contain `https`
- url Host is ""
- in.config.RegionName != regionName

May be this is overkill for the situation , if @andrewkroh or you has any other opinion

Reply:
I've reworked this. Will send soon.